compiler/passes/src/type_checking/check_program.rs

```diff
@@ -121,6 +121,21 @@ impl<'a> ProgramVisitor<'a> for TypeChecker<'a> {
             };
             check_has_field(sym::owner, Type::Address);
             check_has_field(sym::gates, Type::U64);
+
+            // Check that the record does not contain another record.
+            for member in input.members.iter() {
+                if let CircuitMember::CircuitVariable(_, Type::Identifier(identifier)) = member {
```

Comment:
Good catch! 
I did not consider the case where a circuit contains a record.
A couple notes:
- Circuits cannot contain records. I missed this case but we will need to add an error for it.
- Tuples are not allowed in circuits or records. This will probably change in the future, so it wouldn't hurt to implement functionality right now.
- I believe it would suffice if we visit each circuit/record declaration and check that its members (traversing tuple types as needed) do not contain a record. We will have to check for recursive types, which is a WIP in #2008.